cmd/kubeadm/app/phases/upgrade/versiongetter.go

```diff
@@ -63,7 +61,7 @@ func (g *KubeVersionGetter) ClusterVersion() (string, *versionutil.Version, erro
 	if err != nil {
 		return "", nil, errors.Wrap(err, "Couldn't fetch cluster version from the API Server")
 	}
-	fmt.Fprintf(g.w, "[upgrade/versions] Cluster version: %s\n", clusterVersionInfo.String())
+	g.printer.Printf("[upgrade/versions] Cluster version: %s\n", clusterVersionInfo.String())
```

Comment:
[1] Ideally, we should get rid of the print inside the getter, and let the caller decide if to print or not the returned value. When this change merges, then the printer parameter can be removed from this func 

But I'm not going to block on this (it can be addressed on follow up PRs as well.

Reply:
Yes, I was thinking about removing this kind of things. However, considering that this function was called from different places I thought it's dangerous to remove the output as in can cause user confusion/ make current UI less informative. Moving it to all places where this function is called doesn't look as a good solution to me either.

I'd prefer to leave it as it is and address this in a separate PRs.

Reply:
Ok, if you open an issue to keep track of this might be we can get some help too (this can make a good first issue)

Reply:
Created an issue: https://github.com/kubernetes/kubeadm/issues/1853